backend/lib/polygon_api.py

```diff
@@ -45,19 +46,17 @@ def get_news(self, ticker: str, max_items: int) -> list[TickerArticle]:
         news_generator: Iterator[TickerNews] | HTTPResponse = self.client.list_ticker_news(
             ticker=ticker, limit=1)
         articles = []
-        for _ in range(max_items):
+        count = 0
+        while count < max_items:
             n = news_generator.__next__()
             # We only get bytes if calling list_ticker_news with raw=True, so can assert TickerNews
             assert isinstance(n, TickerNews)
-
-            assert n.title is not None
-            assert n.description is not None
-            assert n.article_url is not None
-            assert n.published_utc is not None
-
-            article = TickerArticle(n.title, n.description,
-                                    n.article_url, n.published_utc)
-            articles.append(article)
+            
+            if (n.title and n.description and n.article_url and n.published_utc):        
```

Comment:
Use this as a guard statement - `assert False` if it fails. This will prevent the server from silently failing, we expect validation to pass so should make noise if it doesn't.

Reply:
What if I want:
- If it doesn't pass, then continue to the next iteration of the while loop (which gets the next news article from the iterator)
- We do this until we've added N valid articles to our list (hence the `while count < max_items`)